crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -251,6 +253,17 @@ pub struct WritebackCache {
     // - note that we removed any unfinalized packages from the cache during revert_state_update().
     packages: MokaCache<ObjectID, PackageObject>,
 
+    // When acquire transaction locks, lock entries are briefly inserted into this map. The map
+    // exists to provide atomic test-and-set operations on the locks. After all locks have been inserted
+    // into the map, they are written to the db, and then all locks are removed from the map.
+    //
+    // After a transaction has been executed, newly created objects are available to be locked.
+    // But, because of crash recovery, we cannot rule out that a lock may already exist in the db for
+    // those objects. Therefore we do a db read for each object we are locking.
+    //
+    // TODO: find a strategy to allow us to avoid db reads for each object.
+    locked_transactions: DashMap<ObjectRef, LockDetails>,
```

Comment:
I think it would be useful to move `locked_transactions` into a struct on its own in a separate file. This struct then contains a DashMap internally. By doing so we could move all the lock related logic there, and have unittests in that file too.

Reply:
Also, probably name it something something _mutex_table is a bit easier to understand/more consistent with previous concepts?